tests/Aspire.Dashboard.Tests/TelemetryRepositoryTests/TraceTests.cs

```diff
@@ -488,4 +489,79 @@ public void AddTraces_AttributeAndEventLimits_LimitsApplied()
         Assert.Equal(5, trace.FirstSpan.Events.Count);
         Assert.Equal(5, trace.FirstSpan.Events[0].Attributes.Length);
     }
+
+    [Fact]
+    public void AddTraces_ExceedLimit_FirstInFirstOut()
+    {
+        // Arrange
+        var repository = CreateRepository(maxTraceCount: 10);
+
+        var testTime = s_testTime.AddDays(1);
+
+        // Act
+        for (var i = 0; i < 2000; i++)
+        {
+            var traceId = (i + 1).ToString(CultureInfo.InvariantCulture);
+
+            // Insert traces out of order to stress the circular buffer type.
+            var startTime = testTime.AddMinutes(i + (i % 2 == 0 ? -5 : 0));
```

Comment:
Should we also try to test for a case where the startTime is before the time of the first entry in the traces at any given time, and maybe after the last one?

Reply:
I looked at code coverage at least some tests hit that path.